packages/tabs/src/vaadin-tabs.js

```diff
@@ -236,12 +236,102 @@ class Tabs extends ResizeMixin(ElementMixin(ListMixin(ThemableMixin(PolymerEleme
 
   /** @private */
   _scrollForward() {
-    this._scroll(-this.__direction * this._scrollOffset);
+    const forwardButtonWidth = this._getNavigationButtonVisibleWidth('forward-button');
+    const backButtonWidth = this._getNavigationButtonVisibleWidth('back-button');
+    const scrollerElementBoundingClientRect = this._scrollerElement.getBoundingClientRect();
+    let itemToScrollTo;
+    for (let i = this.items.length - 1; i >= 0; i--) {
+      const isItemVisible = this._isItemVisible(
+        this.items[i],
+        true,
+        forwardButtonWidth,
+        backButtonWidth,
+        scrollerElementBoundingClientRect,
+      );
+      if (isItemVisible) {
+        itemToScrollTo = this.items[i];
+        break;
+      }
+    }
+    const itemBoundingClientRect = itemToScrollTo.getBoundingClientRect();
+    const transitionCompensation = 20;
+    let scrollOffset;
+    if (this.__isRTL) {
+      const scrollerRightEdge = scrollerElementBoundingClientRect.right - backButtonWidth - transitionCompensation;
+      scrollOffset = itemBoundingClientRect.right - scrollerRightEdge;
+    } else {
+      const scrollerLeftEdge = scrollerElementBoundingClientRect.left + backButtonWidth + transitionCompensation;
+      scrollOffset = itemBoundingClientRect.left - scrollerLeftEdge;
+    }
+    if (Math.floor(scrollOffset) === 0) {
+      scrollOffset = -this.__direction * (this._scrollOffset - backButtonWidth - transitionCompensation);
+    }
+    this._scroll(scrollOffset);
   }
 
   /** @private */
   _scrollBack() {
-    this._scroll(this.__direction * this._scrollOffset);
+    const forwardButtonWidth = this._getNavigationButtonVisibleWidth('forward-button');
+    const backButtonWidth = this._getNavigationButtonVisibleWidth('back-button');
+    const scrollerElementBoundingClientRect = this._scrollerElement.getBoundingClientRect();
```

Comment:
I'd suggest simplifying the naming of variables that store the result of `getBoundingClientRect()` considering that the method doesn't use any other methods to calculate rectangles.

```suggestion
    const scrollerRect = this._scrollerElement.getBoundingClientRect();
```

```
itemRect
```

Reply:
Makes sense. I renamed all the variables that match this case.